### PR TITLE
GAWB-580: Markdown rendering for description

### PR DESCRIPTION
**Pull request by MattPutnam**
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] **Submitter**: Tell tech lead that the PR exists if s/he wants to look at it
- [x] **Submitter**: Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x] **Tech lead**: sign off
- [x] **LR**: sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't 
  working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev server still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/org/broadinstitute/firecloud_ui/common/markdown.cljs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    ))
+
+;; Documentation:
+;; https://github.com/chjj/marked
+
+(js/marked.setOptions
+  #js{:sanitize true})
+
+(defonce ^:private renderer (js/marked.Renderer.))
+(set! renderer.link
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/common/markdown.cljs]:
``` clojure
(set! (.-link renderer) ...)
```

would be a bit more idiomatic.

File: src/cljs/org/broadinstitute/firecloud_ui/common/markdown.cljs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  (fn [href, title, text]
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/common/markdown.cljs]: Let's toss these commas.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/tab.cljs [hunk 3]
```diff
@@ -131,53 +132,54 @@
...
+               (for [[status subs] (sort submissions-count)]
+                 [:li {} (str subs " " status)])])]))]]
+     (style/create-section-header "Description")
+     (style/create-paragraph
+       (let [description (not-empty (get-in ws ["workspace" "attributes" "description"]))]
+         (cond (:editing? @state) (react/create-element
+                                    (style/create-text-area {:ref "descriptionArea"
+                                                             :defaultValue description
+                                                             :style {:width "100%"}
+                                                             :rows 10}))
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/tab.cljs:179]: I would love for us to come up with a textarea control that expanded its rows depending on its content up to some maximum value. If you think that would be useful too, would you create a story?

File: src/static/assets/css/markdown.css [hunk 4]
```diff
@@ -0,0 +1,695 @@
+@font-face {
```
Comment (dmohs) [inline src/static/assets/css/markdown.css]: I would prefer it if we didn't store external code in our repo, but I would understand if you think that is outside the scope of this PR. If so, would you create a story to address this?

**Comment by dmohs** [comment]
👍 Minor suggestions. Looks good.

File: src/static/assets/css/markdown.css (hunk 4 above)
Comment (MattPutnam) [inline src/static/assets/css/markdown.css]: I don't think we need to track changes to this file.  Effectively, it's our file that we've just cribbed from an external source.

Comment (davidangb) [inline src/static/assets/css/markdown.css]: From where did you get this CSS? What is its license? Can we just link to the external file, hosted somewhere else?If not, add a comment explaining where we got it.

**Comment by davidangb** [comment]
See comment inline about license/source for CSS.

I was going to suggest that you encode the link title/text as well as the href inside the renderer ... but I couldn't manage to inject anything. Would encoding hurt here? (would it be double-encoded?)

File: src/static/assets/css/markdown.css (hunk 4 above)
Comment (MattPutnam) [inline src/static/assets/css/markdown.css]: https://github.com/sindresorhus/github-markdown-css

MIT license.  We can't just link it because I've already changed things--the link color and not setting its own font size.  I'll add a comment.

Comment (davidangb) [inline src/static/assets/css/markdown.css]: Would it be possible to link it as-is, then override some changes using `!important` or some other technique? Or, is there some clsjs/leinengen -> to -> npm bridge that would allow us to install via dependency management at build time?

If not, please add a NOTICE file according to the Broad's third-party licensing policy here: https://broadinstitute.atlassian.net/wiki/display/DSDE/DSDE+Open+Source+Licensing#DSDEOpenSourceLicensing-Redistributing3rdPartyLibraries

Even if we do need to check it in directly, I'd like to explore checking it in as-is and overriding with `!important` (et al), to ease any upgrades in the future.

File: script/start-hot-reloader.sh [hunk 5]
```diff
@@ -10,5 +10,5 @@ exec docker run --rm -it \
   -w /work -v "$PWD":/work -v "$HOME"/.m2:/root/.m2 \
   -e FIGWHEEL_HOST="${FIGWHEEL_HOST:-}" \
   -p 3449:3449 \
-  dmohs/clojure \
+  clojure \
```
Comment (MattPutnam) [inline script/start-hot-reloader.sh:13]: This fixed a bug I encountered while trying to reload the css files.

**Comment by davidangb** [comment]
👍 !!
